sui_core/src/authority_client.rs

```diff
@@ -7,9 +7,12 @@ use async_trait::async_trait;
 use futures::{stream::BoxStream, TryStreamExt};
 use multiaddr::Multiaddr;
 use std::sync::Arc;
+
```

Comment:
Why does the authority client need to handle checkpoints? (maybe I missed something important here)

Reply:
The checkpoint logic is client driven: but of course the client will be the authority and other authorities. So in particular:
- a client seeks k proposals, and makes fragments out of them (one of them from own authority).
- a client submits the fragments to consensus.
- after consensus a client asks for the signed checkpoints
- makes a certificate and submits it to all authorities.

Reply:
Hey @asonnino, I wrote a test that illustrates the process above:
https://github.com/MystenLabs/sui/blob/f72949bbafdd2fb2884454697b5fc3e7be94bfc2/sui_core/src/unit_tests/checkpoint_tests.rs#L1437-L1549